enhancements/compound-bundle-constraints.md

```diff
@@ -0,0 +1,350 @@
+There are two components to a bundle: its Kubernetes manifests, and metadata about those
+manifests such as package name, version, and what group-version-kind/APIs's the manifests provide.
+These metadata are referred to as bundle [properties][properties], and are expressed as [declarative configuration][dc].
+Operators can depend on other operators for providing APIs, ex. those created by CRDs,
+to interact with; a great example is an operator requiring HA config storage and service discovery
+could use Etcd's set of APIs, which are provided by an Etcd operator. These dependencies
+are specified as bundle properties as well, ex. `olm.gvk.required`.
+
+When [OLM][olm] installs a bundle with dependency properties, the [resolver][olm-resolver] will evaluate them,
+reduced down to a boolean expression, over the non-dependency properties of all currently installed bundles.
+Because there is a difference between properties that can and cannot be evaluated,
+the former will be referred to as **constraints** in this EP; specifically, these
+are dependency constraints. After the resolver completes evaluation,
+it either returns a non-satisfiability error or the "best" choice of dependency bundle to install.
+
+## Motivation
+
+The existing constraints `olm.package.required` and `olm.gvk.required` are the only constraints
+required for most use cases. They are specified as follows:
+
+```yaml
+---
+schema: olm.bundle
+name: foo.v1.0.0
+package: foo
+properties:
+- type: olm.package.required
+  value:
+    packageName: bar
+    versionRange: '>=1.0.0'
+- type: olm.gvk.required
+  value:
+    group: etcd.database.coreos.com
+    kind: EtcdBackup
+    version: v1beta2
+```
+
+The above bundle `foo.v1.0.0` requires that both a bundle in package `bar` that is of at least
+version 1.0.0 _and_ a bundle that provides the `EtcdBackup` API in group `etcd.database.coreos.com`
+and at version `v1beta2`. The resolver will translate those constraints into a boolean expression
+that is evaluated over all installed bundles's properties in the cluster; if both `bar` and `etcd`
+bundles with properties satisfying group and/or version constraints are installed, then
+`foo.v1.0.0` installation can proceed.
+
+However, what if:
+- `foo.v1.0.0` can use multiple versions of the same API, but only one of that group-kind?
+- `foo.v1.0.0` can use multiple versions of the same API, but only if another package version is present?
+- `foo.v1.0.0` cannot use a particular API version, but can use all others?
+
+These constraints are logical disjunctions and negations in character; the existing
+top-level constraints are together a logical conjunction, although not explicit.
+Currently the spec does not support specifying such "compound" dependency constraints.
+
+### Goals
+
+- Define logical conjunction, disjunction, and negation properties as dependency constraints, termed "compound constraints".
+- Define nesting of compound constraints constraints.
```

Comment:
I'm assuming this does not cover 'soft dependencies' where an operator might be installed in ways that would dynamically alter its dependencies?  Is that out of scope for this?

Reply:
I think there might be a typo in this line as well "constraints constraints"